src/core/channel.c

```diff
@@ -115,6 +115,12 @@ int core_channel_size(core_channel_t* self)
     return ck_ring_size(&self->ring);
 }
 
+bool core_channel_full(core_channel_t* self)
+{
+    mlassert_self();
+    return (core_channel_size(self) >= (self->capacity - 1));
```

Comment:
I don't get why you do `- 1` here, `ck_ring_size()` returns the number of enqueued pointers and if capacity is 2 then 1 enqueued would mean full.

Reply:
Sorry for noticing this earlier. From `man ck_ring_capacity`:
> Note that a ring can only hold ck_ring_capacity(3) minus one entries at a time.

Thus when there are 3 items in a 4-capacity ring buffer, it is no longer possible to enqueue any more items, and it should be considered full.

Simple test case:
```
#include <ck_ring.h>
#include <stdio.h>
#include <assert.h>

void main() {
	int capacity = 4;
	ck_ring_t ring;
	ck_ring_buffer_t buffer[capacity];

	ck_ring_init(&ring, capacity);
	assert(ck_ring_capacity(&ring) == capacity);

	bool ret;
	int size;
	for (int i = 0; i < ck_ring_capacity(&ring); i++) {
		ret = ck_ring_enqueue_spsc(&ring, buffer, 0);
		size = ck_ring_size(&ring);
		printf("i=%d: %s (size=%d)\n", i, ret ? "ok" : "not ok", size);
	}
}
```

outputs:
```
i=0: ok (size=1)
i=1: ok (size=2)
i=2: ok (size=3)
i=3: not ok (size=3)
```